docs/project-model/project-json-reference.md

```diff
@@ -0,0 +1,847 @@
+
+For example:
+
+    {
+        "compilationOptions": {
+            "emitEntryPoint": true
+        }
+    }
+
+<a name="xmlDoc"></a>
+### xmlDoc
+Type: Boolean
+
+Enables xml docs to be generated from comments in the sourcecode.
+
+For example:
+
+    {
+        "compilationOptions": {
+            "xmlDoc": true
+        }
+    }
+
+<a name="preserveCompilationContext"></a>
+### preserveCompilationContext
+Type: Boolean
+
+TODO
+
+<a name="analyzerOptions"></a>
+## analyzerOptions
+Type: Object
+
+An object with properties used by code analysers.
+
+For example:
+
+    {
+        "analyzerOptions": { }
+    }
+
+<a name="languageId"></a>
+### languageId
+Type: String
+
+TODO: Check language ids
+The id of the language to analyze.
+
+For example:
+
+    "analyzerOptions": {
+      "languageId": "vb"
+      }
+    }
+
+<a name="configurations"></a>
+## configurations
+Type: Object
+
+An object whos properties define different configurations for this project, such as Debug and Release. Each value is an object that can contain a compilationoptions object with options specific for this coniguration.
+
+For example:
+
+    "configurations": {
+      "Release": {
+        "compilationOptions": {
+          "allowUnsafe": false
+        }
+      }
+    }
+
+<a name="frameworks"></a>
+## frameworks
+Type: Object
+
+Specifies what frameworks this project supports, such as .NET Framework or Universal Windows Platform (UWP). Must be a valid Target Framework Moniker (TFM). Each value is an object that can contain information specific to this framework such as `compilationoptions`, `analyzeroptions` as well as the following properties.
+
+For example:
+
+    "dependencies": {
+        "Microsoft.Extensions.JsonParser.Sources": {
+            "type": "build",
+            "version": "1.0.0-rc2-16453"
+        },
+        "Microsoft.Extensions.HashCodeCombiner.Sources": {
+            "type": "build",
+            "version": "1.0.0-rc2-16054"
+        },
+        "Microsoft.Extensions.DependencyModel": "1.0.0-*"
+    },
+
+<a name="dependencies"></a>
+### dependencies
+Type: Object
+
+Dependencies that are specific for this framework. This is useful in scenarios where you cannot simply specify a package-level dependency across all targets. Reasons for this can include one target lacking built-in support that other targets have, or requiring a different version of a dependency than other targets.
+
+For example:
+
+    "frameworks": {
+        "dnxcore50": {
+            "dependencies": {
+                "NETStandard.Library": "1.0.0-rc2-23811",
+            }
+        }
+    }
+
+<a name="frameworkAssemblies"></a>
+### frameworkAssemblies
+Type: Object
+
+Similar to dependencies but contains reference to assemblies in the GAC that are not nuget packages. Can also specify the version to use as well as the dependency type. Only used by the .NET Framework and Portable Class Library (PCL) references.
```

Comment:
> contains reference to assemblies in the GAC that are not nuget packages

This is actually something [we're looking to change](https://github.com/dotnet/corefx/issues/3934#issuecomment-179963211).  That effort is blocked right now, so I think it's okay to state that this is for GAC reference assemblies for the time being.